fletcher/algorithms/string.py

```diff
@@ -267,6 +268,72 @@ def _text_contains_case_sensitive(data: pa.Array, pat: str) -> pa.Array:
     )
 
 
+@njit
+def _get_zstring(width: int, buffer: np.ndarray) -> np.ndarray:
+    num_zeroes = width
+    for i in np.bitwise_xor(np.right_shift(buffer, 6), 2):
+        if i != 0:
+            num_zeroes -= 1
+    if num_zeroes <= 0:
+        return buffer
+    # Create array of byte-representation of 0s concat'd with buffer
+    return np.concatenate((np.full(num_zeroes, 48, dtype=np.uint8), buffer), axis=0)
+
+
+@njit
+def _zfill_nonnull(
+    length: int, offsets: np.ndarray, data_buffer: np.ndarray, width: int
+):
+    str_builder = StringArrayBuilder(max(len(data_buffer), 2))
```

Comment:
Also a style/intuition question @xhochy : is there a preference for passing in `str_builder` as a parameter of `_zfill_nonnull` and `_zfill_nulls` (i.e. initializing it within the calling function `_zfill`) vs. initializing it within each of these functions? Benchmarks don't appear to make a noticeable difference, and I do like eliminating the duplicated line by passing it in, but wasn't sure if there's a style preference for one over the other. Thanks!

Reply:
I would also prefer to pass it in from the outside.